lib/mix/tasks/git_ops.release.ex

```diff
@@ -78,51 +83,59 @@ defmodule Mix.Tasks.GitOps.Release do
         """
       end
 
+    repo = GitOps.Git.init!()
+
     opts = get_opts(args)
 
     if opts[:initial] do
-      GitOps.Changelog.initialize(path)
-    end
-
-    if not File.exists?(path) do
-      raise "\nFile: #{path} did not exist. Please use the `--initial` command to initialize."
+      GitOps.Changelog.initialize(changelog_path)
     end
 
     tags = GitOps.Git.tags(repo)
 
     prefix = GitOps.Config.prefix()
 
+    config_types = GitOps.Config.types()
+
     {commit_messages_for_version, commit_messages_for_changelog} =
       get_commit_messages(repo, prefix, tags, opts)
 
-    config_types = GitOps.Config.types()
-
     commits_for_version = parse_commits(commit_messages_for_version, config_types, true)
     commits_for_changelog = parse_commits(commit_messages_for_changelog, config_types, false)
 
-    prefixed_version =
+    prefixed_new_version =
       if opts[:initial] do
-        mix_project_module.project()[:version]
+        mix_project[:version]
       else
-        GitOps.Version.determine_new_version(tags, prefix, commits_for_version, opts)
+        GitOps.Version.determine_new_version(
+          current_version,
+          prefix,
+          commits_for_version,
+          opts
+        )
       end
 
-    new_version = String.trim_leading(prefixed_version, prefix)
+    new_version = String.trim_leading(prefixed_new_version, prefix)
 
-    GitOps.Changelog.write(path, commits_for_changelog, current_version, prefixed_version)
+    if confirm_and_tag(repo, prefixed_new_version) do
```

Comment:
So confirm_and_tag will commit the changes, but this writes the changelog *after* the commit step. I imagine what we'd want to do instead is to get confirmation, and then write the changelog, commit, and tag. Unless this is supposed to be a step for revising the changelog? If it is, then we should say something like: please review the changelog, add any details you feel are necessary, and commit that with `docs: update changelog` or something along those lines.

Reply:
Yeah.. I was toying around with how this would be done. At first I was thinking of displaying the changes to the user, then perhaps we could write them to a temporary changelog file. Unsure exactly what would be best.